src/Verify.MSTest/Verifier.cs

```diff
@@ -0,0 +1,54 @@
+namespace VerifyMSTest;
+
+public static partial class Verifier
+{
+    static InnerVerifier BuildVerifier(string sourceFile, VerifySettings settings)
+    {
+        Guard.AgainstBadSourceFile(sourceFile);
+
+        var method = GetTestMethod();
+        var type = method.DeclaringType!;
+        TargetAssembly.Assign(type.Assembly);
+
+        GetFileConvention fileConvention = (uniquenessForReceived, uniquenessForVerified) =>
+            ReflectionFileNameBuilder.FileNamePrefix(method, type, sourceFile, settings, uniquenessForReceived, uniquenessForVerified);
+
+        return new(sourceFile, settings, fileConvention);
+    }
+
+    static MethodInfo GetTestMethod()
+    {
+        var stackTrace = new StackTrace();
```

Comment:
stack walking is unreliable and very diffcult to debug. so this approach is not viable

Reply:
which other approach do you suggest then?

Reply:
i dont know. i dont think there is a way to solve it gracefully in MStest.

i think the only to is to pass the extra context through as a parameter to the static Verify method

Reply:
I'm wondering if walking the stack is unreliable in this case. The unit test method will always exist in the assembly, and can never be inlined.

Reply:
it was the approach we use in ApprovalTests. many bugs over many years. Still a few open https://github.com/approvals/ApprovalTests.Net/issues/405 https://github.com/approvals/ApprovalTests.Net/issues/449

Reply:
alright thanks, will need to solve it another way then, since we already have another base class for out tests that is not extendable.